base/string_utils.cpp

```diff
@@ -257,6 +257,18 @@ bool ReplaceLast(std::string & str, std::string const & from, std::string const
   return true;
 }
 
+bool ReplaceAll(std::string & str, std::string const & from, std::string const & to)
+{
+  size_t pos = str.rfind(from);
+  bool out = false;
+  while (pos != std::string::npos) {
+    str.replace(pos, from.length(), to);
+    out = true;
+    pos = str.rfind(from);
+  }
+  return out;
+}
```

Comment:
Will have an infinite loop if _from_ is a substring of _to_.
ReplaceAll("xxyyzz", "yy", "xyyz");
More natural code is like this:
```
size_t pos = 0;
bool out = false;
while ((pos = str.find(from, pos)) != std::string::npos)
{
  str.replace(pos, from.length(), to);
  pos += to.length();
  out = true;
}
return out;
```

Reply:
https://en.cppreference.com/w/cpp/regex/regex_replace with a const static allocated regex can be a better solution.
Replacing many occurrences leads to many reallocations.

Reply:
Undid string_utils change, using regex_replace